### ISSUE TITLE
Clarify that keys of Startup Code json should be kernel display name

### ISSUE THREAD

**Issue by kylebarron** [nteract/hydrogen#1429]
I had been confused for a while why my startup code wasn't running. I had had 
```json
{"python": "%matplotlib inline"}
```

I realize now that the key needs to be the _display name_ of the kernel, so this has to be `Python 3`. This should be clarified in the documentation, because the display name is not always straightforward. For example, the Julia display name (as of 1.0) is `Julia 1.0.0` and the IJavascript display name is `Javascript (Node.js)`. 

Is there any cleaner way to see the display names of installed jupyter kernels other than `jupyter kernelspec list --json` and looking at `display_name` values?

**Comment by BenRussert** [comment]
I totally agree this needs improvement. I'm thinking the display name key was used to be as specific as possible on which kernel should run each startup code.

**Comment by n-riesco** [comment]
@kylebarron It used to be the language name, but we changed it to display names in #195 to address [the last bullet point in this comment](https://github.com/nteract/hydrogen/pull/195#issuecomment-196759817).

How about accepting both possibilities: display and grammar names. Hydrogen first checks if there is an entry for the grammar name. If there isn't, then checks the display name.

**Comment by kylebarron** [comment]
By "grammar name", I'm assuming you mean the key for `language` in the kernelspec JSON?

I don't know of any language for which there are more than one kernel and where the display name matches the language name exactly. If this is the case, we could allow both language names and display names as keys in the startup code JSON. Then when a key is a display name, apply it only for that kernel, and when a key is a language name, apply it for all kernels of that language?

**Comment by n-riesco** [comment]
> By "grammar name", I'm assuming you mean the key for language in the kernelspec JSON?

No, I mean atom's grammar name (Hydrogen should use only atom's grammar names, otherwise it's too confusing; we had that problem a long time ago). 

> Then when a key is a display name, apply it only for that kernel, and when a key is a language name, apply it for all kernels of that language?

Oh yes that makes more sense: display names should have priority over grammar names (as display names are more specific than grammar names).

**Comment by kylebarron** [comment]
> No, I mean atom's grammar name (Hydrogen should use only atom's grammar names, otherwise it's too confusing; we had that problem a long time ago).

So would you want the keys to be `source.python`?

**Comment by n-riesco** [comment]
See [grammarToLanguage](https://github.com/nteract/hydrogen/blob/9d595258c9619b3b106b796c7b1305e3109bcf9e/lib/utils.js#L58-L69) and the flow declaration of [atom$Grammar](https://github.com/nteract/hydrogen/blob/9d595258c9619b3b106b796c7b1305e3109bcf9e/types/atom.js.flow#L1512-L1516) to see what I mean by grammar language. E.g. for Python is just `python` and for JavaScript is `javascript`.

**Comment by kylebarron** [comment]
> It used to be the language name, but we changed it to display names in #195 to address [the last bullet point in this comment](https://github.com/nteract/hydrogen/pull/195#issuecomment-196759817).

I agree that startup code should be specific to a kernel and not to a language. You'd possibly have different startup code for Python 2 vs Python 3.

I'd argue that the best use for the key would be the _kernelspec_ name. This is easier for users to find. I.e. just running `jupyter kernelspec list` shows them the list of keys they could use:
```
> jupyter kernelspec list
Available kernels:
  bash          /Users/kyle/Library/Jupyter/kernels/bash
  ir            /Users/kyle/Library/Jupyter/kernels/ir
  javascript    /Users/kyle/Library/Jupyter/kernels/javascript
  julia-0.6     /Users/kyle/Library/Jupyter/kernels/julia-0.6
  stata         /Users/kyle/Library/Jupyter/kernels/stata
  python3       /Users/kyle/anaconda/share/jupyter/kernels/python3
```
As of https://github.com/nteract/hydrogen/pull/1499 the kernelspec name is saved in `store`. We'd probably want to keep display name as a fallback for backwards compatibility.
